gsi/hnswlib_test/gxl.py

```diff
@@ -0,0 +1,76 @@
+import hnswlib
+import numpy as np
+import pandas as pd
+import datetime
+import os
+
+k = 10
+dim = 96
+ef_search = [64, 128, 256, 512]
+results = []
+
+index_dir = '/mnt/nas1/GXL/deep1B/v2.0_with250Mfix/'
+filename = 'deep1B_250m_ef_64_M_32_gxl.bin'
+index_path = os.path.join(index_dir, filename)
+#data_path = '/home/gwilliams/Projects/GXL/deep-10M.npy'
+data_path = '/mnt/nas1/fvs_benchmark_datasets/deep-250M.npy'
+#index_path = '/home/gwilliams/Projects/GXL/deep1B_50m_ef_64_M_32_gxl.bin'
+query_path = '/home/gwilliams/Projects/GXL/deep-queries-1000.npy'
+queries = np.load(query_path, allow_pickle=True)
+
+def size_num(s):
+    '''get raw numercs of text abbrev'''
+    if s == '1m': return 1000000
+    elif s == '2M': return 2000000
+    elif s == '5M': return 5000000
+    elif s == '10m': return 10000000
+    elif s == '20m': return 20000000
+    elif s == '50m': return 50000000
+    elif s == '100m': return 100000000
+    elif s == '200M': return 200000000
+    elif s == '250m': return 250000000
+    elif s == '500m': return 500000000
+    elif s == '10K': return 10000
+    elif s == '1000m': return 1000000000
+    else: raise Exception("Unsupported size " + s)
+
+# parse parameters
```

Comment:
put comment for this section of code - such as "we parse the GXL index file name for various parameters"